test/run

```diff
@@ -622,7 +663,11 @@ echo
 cd $TESTDIR || exit 1
 
 mkdir compiler
+
 COMPILER="$(pwd)/compiler/$(basename "$REAL_COMPILER_BIN")"
+if $HOST_OS_WINDOWS; then 
+    COMPILER="$COMPILER.sh"
```

Comment:
I don't quite understand why shell scripts need to have a `.sh` extension. Is that a requirement in bash in MSYS2? Hmm, or is it because of ccache's strange usage of `Win32::add_exe_prefix`? If it's the latter, would it be possible to fix that instead, i.e., only add `.exe` to the path of the path doesn't already exist?

Reply:
Done

Reply:
As far as I remember, the .sh extension was necessary, otherwise the script was not started in the shell. If the gitbash is installed, *.sh files are assigned to it.

Alternatively you would have to extend ccache to evaluate the shebang and then run the script with the matching executable.

Reply:
It didn't work. See [this run](https://github.com/ccache/ccache/runs/7906502881?check_suite_focus=true).

Reply:
Reverted to adding sh extension, and removed the commit that modifies `add_exe_suffix`.